src/main/scala/io/iohk/ethereum/network/MaxBlockNumberRequestHandler.scala

```diff
@@ -0,0 +1,64 @@
+package io.iohk.ethereum.network
+
+import akka.actor.{ActorRef, Terminated}
+import io.iohk.ethereum.network.BlockBroadcastActor.ProcessingState
+import io.iohk.ethereum.network.MaxBlockNumberRequestHandler.{PeerTerminated, RequestMaxBlockNumber}
+import io.iohk.ethereum.network.PeerActor.{GetMaxBlockNumber, MaxBlockNumber}
+import io.iohk.ethereum.network.p2p.messages.CommonMessages.NewBlock
+
+//FIXME: Decide block propagation algorithm (for now we send block to every peer except the sender) [EC-87]
+trait MaxBlockNumberRequestHandler { this: BlockBroadcastActor =>
```

Comment:
You are right, it was misleading. I just changed this trait's name to `BlockSendingHandler`